### PR TITLE
Add IdentityApi to check rewards eligibility

### PR DESCRIPTION
**Pull request by hectorgomezv**
## Summary
This PR implements an eligibility check for the Safe clients. The main criterion for eligibility (e.g.: eligible for rewards) is the country of origin of the incoming request. 

For that purpose, FingerprintJS is used. This service requires the client to install a JS Agent, that makes a request to Fingerprint servers. The response contains a _sealed data_ payload, holding **anonymous** identity data like browser type, VPN information, IP location, etc. Then, this data can be sent to the CGW, which _unseals_ it, and applies the eligibility logic.

A more detailed explanation of how this works can be found in the [Fingerprint documentation for Sealed Client Results](https://dev.fingerprint.com/docs/sealed-client-results).

## Changes
- Adds a `CommunityRepository['checkEligibility']` function to check if a specific request is eligible.
- Adds `FingerprintApiService` that wraps a `FingerprintJS` library, to unseal the request identity payload.
- Adds `FINGERPRINT_NON_ELIGIBLE_COUNTRIES` as a non-mandatory environment variable (defaulting to `['US']`).
- Adds `FINGERPRINT_ENCRYPTION_KEY` as a mandatory environment variable.

## Out of scope
- It could be different eligibility conditions per community campaign. The PR assumes all the campaigns share the same eligibility conditions.
- It could be other eligibility conditions besides the request origin country. The PR assumes the only criterion is the origin country.

### CODE REVIEW

File: src/config/entities/configuration.ts [hunk 1]
```diff
@@ -247,6 +247,11 @@ export default () => ({
     baseUri:
       process.env.LOCKING_PROVIDER_API_BASE_URI ||
       'https://safe-locking.safe.global',
+    eligibility: {
+      fingerprintEncryptionKey: process.env.FINGERPRINT_ENCRYPTION_KEY,
+      nonEligibleCountries:
+        process.env.FINGERPRINT_NON_ELIGIBLE_COUNTRIES?.split(',') ?? ['US'],
```
[the remark this answers is not in the document]
Comment (hectorgomezv) [reply src/config/entities/configuration.ts]: Changed in [a85432f](https://github.com/safe-global/safe-client-gateway/pull/2103/commits/a85432fa231dbf6cd36656d390880abcd522a7d1)

File: src/datasources/locking-api/fingerprint-api.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    service = new FingerprintApiService(
+      fakeConfigurationService,
+      mockLoggingService,
+    );
+  });
+
+  describe('checkEligibility', () => {
+    it('should return isAllowed:true and isVpn:false for a non-VPN non-US location', async () => {
+      const eligibilityRequest = eligibilityRequestBuilder().build();
+      const unsealedData = {
```
[the remark this answers is not in the document]
Comment (hectorgomezv) [reply src/datasources/locking-api/fingerprint-api.service.spec.ts]: I've added the type, schema (with tests), and its test entity builders in [a85432f](https://github.com/safe-global/safe-client-gateway/pull/2103/commits/a85432fa231dbf6cd36656d390880abcd522a7d1)

File: src/datasources/locking-api/fingerprint-api.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+      expect(unsealEventsResponse).toHaveBeenCalledWith(
+        Buffer.from(eligibilityRequest.sealedData, 'base64'),
+        [
+          {
+            key: Buffer.from(eligibilityEncryptionKey, 'base64'),
+            algorithm: 'aes-256-gcm',
+          },
+        ],
+      );
```
[the remark this answers is not in the document]
Comment (hectorgomezv) [reply src/datasources/locking-api/fingerprint-api.service.spec.ts]: I removed these blocks in [a85432f](https://github.com/safe-global/safe-client-gateway/pull/2103/commits/a85432fa231dbf6cd36656d390880abcd522a7d1)

File: src/datasources/locking-api/fingerprint-api.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+      expect(result).toEqual({
+        requestId: eligibilityRequest.requestId,
+        isAllowed: false,
+        isVpn: false,
+      });
+    });
```
[the remark this answers is not in the document]
Comment (hectorgomezv) [reply src/datasources/locking-api/fingerprint-api.service.spec.ts]: Removed in [a85432f](https://github.com/safe-global/safe-client-gateway/pull/2103/commits/a85432fa231dbf6cd36656d390880abcd522a7d1)

**Review by iamacook — APPROVED** [review]

**Pull request merged**